domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterControllerTest.kt

```diff
@@ -0,0 +1,260 @@
+    testCoroutineDispatchers.runCurrent()
+
+    // The platformParameterMap must have values as application had cached platform parameter data.
+    verify(mockUnitObserver, atLeastOnce()).onChanged(unitCaptor.capture())
+    assertThat(unitCaptor.value.isSuccess()).isTrue()
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isNotEmpty()
+    checkEntriesInsidePlatformParameterMap(platformParameterSingleton.getPlatformParameterMap())
+  }
+
+  @Test
+  fun testController_noPreviousDatabase_updatePlatformParameterDatabase_readPlatformParameters_platformParameterMapHasValues() { // ktlint-disable max-line-length
+    setUpTestApplicationComponent()
+    platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
+    testCoroutineDispatchers.runCurrent()
+    platformParameterController.getParameterDatabase().toLiveData().observeForever(mockUnitObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // The platformParameterMap must have values as we updated the database with dummy list.
+    verify(mockUnitObserver, atLeastOnce()).onChanged(unitCaptor.capture())
+    assertThat(unitCaptor.value.isSuccess()).isTrue()
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isNotEmpty()
+    checkEntriesInsidePlatformParameterMap(platformParameterSingleton.getPlatformParameterMap())
+  }
+
+  @Test
+  fun testController_existingDatabase_updatePlatformParameterDatabase_readPlatformParameters_platformParameterMapHasNewValues() { // ktlint-disable max-line-length
+    // Simulate that previous app already has cached platform parameter values in cache store.
+    executeInPrevious { testComponent ->
+      testComponent.getPlatformParameterController().updatePlatformParameterDatabase(
+        mockPlatformParameterList
+      )
+      testComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+
+    // Create the application after previous arrangement to simulate a re-creation.
+    setUpTestApplicationComponent()
+    platformParameterController.updatePlatformParameterDatabase(listOf())
+    testCoroutineDispatchers.runCurrent()
+    platformParameterController.getParameterDatabase().toLiveData().observeForever(mockUnitObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // The new set of values must be empty as we updated the database with an empty list.
+    verify(mockUnitObserver, atLeastOnce()).onChanged(unitCaptor.capture())
+    assertThat(unitCaptor.value.isSuccess()).isTrue()
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+  }
+
+  /**
+   * This function checks does all the entries inside the [mockPlatformParameterList] exist inside
+   * [platformParameterMap] that was retrieved from cache store.
+   * @param platformParameterMap Map<String, PlatformParameter> map of cached values
+   * */
+  private fun checkEntriesInsidePlatformParameterMap(
```

Comment:
`verifyEntriesInsidePlatformParameterMap` -- I think we generally use `verify` to indicate we are validating something.

Reply:
Done